test/idp_fraud_data_handler_test.py

```diff
@@ -0,0 +1,750 @@
+                        idp_fraud_events[2].idp_event_id
+                    )
+                ),
+                (
+                    'idp_fraud_data_handler',
+                    'WARNING',
+                    'Successfully wrote IDP fraud event ID {} to database BUT no matching fraud event found'.format(
+                        idp_fraud_events[3].idp_event_id
+                    )
+                ),
+                (
+                    'idp_fraud_data_handler',
+                    'ERROR',
+                    'Failed to store IDP fraud event: list index out of range (line 6)'
+                ),
+                (
+                    'idp_fraud_data_handler',
+                    'WARNING',
+                    'Processing Failed'
+                ),
+            )
+            self.__assert_upload_session_exists_in_database(False)
+            self.__assert_events_exist_in_database(idp_fraud_events)
+            self.__assert_error_in_database_failure_table(
+                6,
+                '**Row Exception**',
+                'Failed to store IDP fraud event: list index out of range (line 6)'
+            )
+            self.__assert_upload_file_has_been_moved_to_folder(idp_fraud_data_handler.ERROR_FOLDER)
+
+    def test_handles_dates_in_different_formats_and_dst(self):
+        idp_fraud_events = self.__generate_test_idp_fraud_events([
+            IdpFraudEvent(
+                timestamp="2019-03-01T02:40:40.1110000Z",
+                idp_event_id="5555555",
+                idp_entity_id=IDP_ENTITY_ID,
+                fid_code="DF01",
+                contra_indicators=["A04", "D02"],
+                contra_score=-5,
+                request_id="_{}".format(uuid.uuid4()),
+                client_ip_address="111.222.222.111",
+                pid=str(uuid.uuid4())
+            ),
+            IdpFraudEvent(
+                timestamp="2019-06-01T03:30:30.2220000Z",
+                idp_event_id="6666666",
+                idp_entity_id=IDP_ENTITY_ID,
+                fid_code="DF01",
+                contra_indicators=["D15"],
+                contra_score=-5,
+                request_id="_{}".format(uuid.uuid4()),
+                client_ip_address="222.111.111.222",
+                pid=str(uuid.uuid4())
+            ),
+        ])
+
+        self.__write_import_file_to_s3(idp_fraud_events)
+
+        with LogCapture('idp_fraud_data_handler', propagate=False):
+            idp_fraud_data_handler.idp_fraud_data_events(self.__create_s3_event(), None)
+
+            self.__assert_upload_session_exists_in_database(True)
+            self.__assert_events_exist_in_database(idp_fraud_events)
+            self.__assert_upload_file_has_been_moved_to_folder(idp_fraud_data_handler.SUCCESS_FOLDER)
+
+    @parameterized.expand([
+        ["comma", ","],
+        ["lf", "\n"],
+        ["crlf", "\r\n"],
+    ])
+    def test_different_delimiters_in_contra_indicator_list(self, name, delimiter):
```

Comment:
We don't currently seem to use the `name` anywhere..?

Reply:
From the docs for `parameterize`, it seems to be required (https://pypi.org/project/parameterized/)

Reply:
It actually works fine. E.g.:

```
    @parameterized.expand([
        ",",
        "\n",
        "\r\n",
    ])
    def test_different_delimiters(self, delimiter):
        assert delimiter in [',', '\n', '\r\n']
```

But admittedly I hadn't realised it was creating new test cases and that the `name` was literally what it was appending to the name of the virtual tests. No name seems to result in tests getting a double underscore at the end: `test_different_delimiters_2__` instead of `test_different_delimiters_2_crlf`.

So feel free to leave as is. 😆